Endpoint Eliminación Categorías:

devmontini opened the pull request:
Se creo la ruta controllador del delete de categories

devmontini committed:
Endpoint Eliminación Categorías:
    Se creo la ruta controllador del delete de categories
    Se agrego una funcion adaptable en dao

soyarielruiz commented on controllers/categories.js:
```diff
@@ -1,3 +1,19 @@
+  static async deleteCategory(req, res) {
+    const { id } = req.params;
+    try {
+      const where = { id: id };
```
nit: Podes hacer `{ id }`

soyarielruiz commented on controllers/categories.js:
```diff
@@ -1,3 +1,19 @@
+      const where = { id: id };
+      const categoryDeleted = await CategoryDao.deleteCategory(where);
+
+      categoryDeleted
```
nit: No tomar como referencia el ternario para manejos de errores.

soyarielruiz commented on dao/categorie.js:
```diff
@@ -0,0 +1,25 @@
+const { Category } = db;
+class CategoryDao {
+  /**
+   * Asynchronously delete a category from the database (table Category)
```
nit: Buena práctica usar JSDocs, pero requiere colocar que retorna.

soyarielruiz commented on dao/categorie.js:
```diff
@@ -0,0 +1,25 @@
+   */
+  static async deleteCategory(where) {
+    try {
+      const categoryData = await Category.findOne({
```
¿Es necesario saber si existe para borrarlo? ¿Esto no lo hace al intentar destruirlo?

soyarielruiz commented on dao/categorie.js:
```diff
@@ -0,0 +1,25 @@
+   * @param {Object} where - To filter by attribute/s Ej: {name: 'ong'}
+   */
+  static async deleteCategory(where) {
+    try {
```
nit: bloque `try` hacerlo más puntual, en este caso tenes 2 puntos posibles de excepción.

soyarielruiz commented on routes/category.js:
```diff
@@ -0,0 +1,13 @@
+
+router.delete(
+  '/:id',
+  RoleMiddleware.isAdminRole,
```
nit: se podría validar que se ingrese por parámetro el id.

devmontini committed:
Merge branch 'development' into OT46-Categorias_endpoint_delete

devmontini committed:
Se soluciono:
Se mejoro el uso de las constantes y los ternarios (controller category)
Se le dio una respuesta el JSDocs (dao category)
Se unifico el pedido de borrar a la base de datos(Destroy)
Se busco un manejo de errores mas puntual (dao Category)
Se creo un validator para el id por params (ruta Category)

devmontini committed:
Merge branch 'development' into OT46-Categorias_endpoint_delete

devmontini committed:
Se cambio el nombre del archivo category routes a categorie
